### PR TITLE
Minor improvements to Sponge implementation

### PR DESCRIPTION
**Pull request by Minecrell**
I made some minor improvements to the Gradle configuration and the main plugin class of the Sponge implementation that remove redundant parts (either because they're default or not needed for Sponge), or simplify the code:

- Use SpongeGradle for setting plugin version from Gradle
- Use Guice to manage configuration
- Use event filter annotation for interact event
- Set plugin description and URL
- Remove some redundant parts in the Gradle config

### CODE REVIEW

File: worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongeWorldEdit.java [hunk 1]
```diff
@@ -141,20 +139,16 @@ public void serverStarted(GameStartedServerEvent event) {
     private boolean ignoreLeftClickAir = false;
 
     @Listener
-    public void onPlayerInteract(InteractBlockEvent event) {
+    public void onPlayerInteract(InteractBlockEvent event, @Root Player spongePlayer) {
```
Comment (DarkArc) [inline worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongeWorldEdit.java]: I did not know you could do this, that's awesome!

File: worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongeWorldEdit.java [hunk 2]
```diff
@@ -59,7 +58,9 @@
 /**
  * The Sponge implementation of WorldEdit.
  */
-@Plugin(id = SpongeWorldEdit.MOD_ID, name = "WorldEdit", version = "%VERSION%")
+@Plugin(id = SpongeWorldEdit.MOD_ID, name = "WorldEdit",
```
Comment (DarkArc) [inline worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongeWorldEdit.java]: So is this necessary with the ID being set in the gradle build script?

File: worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongeWorldEdit.java [hunk 3]
```diff
@@ -83,7 +84,11 @@ public static SpongeWorldEdit inst() {
     }
 
     private SpongePlatform platform;
+
+    @Inject
     private SpongeConfiguration config;
+
+    @Inject @ConfigDir(sharedRoot = false)
     private File workingDir;
```
Comment (DarkArc) [inline worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongeWorldEdit.java]: Will this automatically run mkdirs?

Comment (Minecrell) [inline worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongeWorldEdit.java]: IIRC this won't but the `@DefaultConfig` injection will.

File: worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongeWorldEdit.java (hunk 2 above)
Comment (Minecrell) [inline worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongeWorldEdit.java]: Well you still need the `@Plugin` annotation with at least the plugin ID. The URL and description could be also defined in the Gradle file but considering it doesn't change every build we can also put it here.

**Comment by DarkArc** [comment]
Merged in fd0e400bcc29016951eb5591f08da106d0f6c359

**Pull request closed without merging**